Modules/Messages/src/Messages.Application/Messages/Commands/SendMessage/Handler.cs

```diff
@@ -75,6 +75,20 @@ private async Task<List<RecipientInformation>> ValidateRecipients(SendMessageCom
                 throw new OperationFailedException(ApplicationErrors.NoRelationshipToRecipientExists(recipientDto.Address));
             }
 
+            var relationshipBetweenSenderAndRecipient = await _relationshipsRepository.FindRelationship(idOfRelationshipBetweenSenderAndRecipient, cancellationToken);
```

Comment:
Now there are two database queries executed. One in line 70 (where only the id is fetched) and another one here (where you fetch the whole relationship). This can easily be done in one query.

Reply:
Yes, sorry.

Changes made.